src/gwc/src/main/java/org/geoserver/gwc/layer/GeoServerTileLayer.java

```diff
@@ -579,57 +587,303 @@ protected final void sendTileRequestedEvent(ConveyorTile tile) {
         }
     }
 
-    protected ConveyorTile getMetatilingReponse(
-            ConveyorTile tile, final boolean tryCache, final int metaX, final int metaY)
+    protected ConveyorTile getMetatilingResponse(
+            ConveyorTile conveyorTile, final boolean tryCache, final int metaX, final int metaY)
             throws GeoWebCacheException, IOException {
 
-        if (tryCache && tryCacheFetch(tile)) {
-            return finalizeTile(tile);
+        if (tryCache && tryCacheFetch(conveyorTile)) {
+            return finalizeTile(conveyorTile);
         }
 
-        final GeoServerMetaTile metaTile = createMetaTile(tile, metaX, metaY);
-        Lock lock = null;
+        final GeoServerMetaTile metaTile = createMetaTile(conveyorTile, metaX, metaY);
+
+        /* ****************** Acquire lock on metatile ******************* */
+        final Lock metaTileLock =
+                GWC.get().getLockProvider().getLock(buildMetaTileLockKey(conveyorTile, metaTile));
```

Comment:
since `GWC.get().getLockProvider().getLock(...)` is a repeating pattern, can we extract it out to a `getLock(String key)` method? (it'd improve readability as I don't need to know how it's acquired everytime while glimpsing over the code)

Reply:
Yes